pkg/metrics/bundledeployment_metrics.go

```diff
@@ -0,0 +1,77 @@
+package metrics
+
+import (
+	fleet "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/rancher/fleet/pkg/summary"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	bundledeploymentSubsystem = "bundledeployment"
+	bundledeploymentLabels    = []string{"name", "namespace", "cluster_name", "cluster_display_name"}
+
+	bundleDeploymentState = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: bundledeploymentSubsystem,
+			Name:      "state",
+			Help:      "Shows the state of this bundle deployment. Ready = 1, NotReady = 2, Pending = 3, OutOfSync = 4, Modified = 5, WaitApplied = 6, ErrApplied = 7.",
```

Comment:
Alternative is to add this as a label instead. Pro being we can now easily query all bundleDeployments in `Ready` state versus `Modified` state instead of figuring out over an indirection what the mapped value looks like. Con being it's a slight increase in cardinality.

Reply:
I previously had this, but noticed that I'd get a record entry for every state change. IE

```
fleet_bundle_ready{state="4" commit="",generation="",name="fleet-agent-c-qls4k",namespace="fleet-default",repo=""} 0
fleet_bundle_ready{state="6" commit="",generation="",name="fleet-agent-c-qls4k",namespace="fleet-default",repo=""} 0
```

So the same bundle had changed state, but now I get a line for each which I would think would be an issue if a bundle was deleted. I could be mis-understanding how this works though.

Reply:
I was expecting the records to look like:
```
fleet_bundledeployment_state{state="ready",commit="",generation="�",name="fleet-agent-c-qls4k",namespace="fleet-default",repo=""} 1
fleet_bundledeployment_state{state="modified",commit="",generation="�",name="fleet-agent-c-qls4k",namespace="fleet-default",repo=""} 0
```

The benefit is it makes querying much easier based on labels since we can aggregate across all the bundleDeployments and identify only the deployments that are in the modified state for more than X minutes (for example) and alert on those.

I mentioned the con being increase in time-series cardinality but I am not too worried about that because it is strongly bounded since each bundleDeployment has a finite (and relatively tiny) limit on number of states. Unbounded labels like `commit` might actually generate more permutations than `state`.

Reply:
Fixed this!